util/chplenv/compiler_utils.py

```diff
@@ -49,3 +64,36 @@ def CompVersion(version_string):
 def compiler_is_prgenv(compiler_val):
   return (compiler_val.startswith('cray-prgenv') or
      os.environ.get('CHPL_ORIG_TARGET_COMPILER','').startswith('cray-prgenv'))
+
+
+#
+# Determine whether a given compiler's default compilation mode
+# supports standard atomics by running the compiler and checking
+# how it expands key feature-test macros.
+#
+# The assumption is that if standard atomics make it into the
+# compiler's default compilation mode, then they actually work.
+# If they are not available in the default mode, they probably
+# have problems and we don't want to use them.
+#
+# Due to the command-line options required, this works for GCC,
+# Clang, and the Intel compiler, but probably not others.
+#
+def has_std_atomics(compiler_val):
+    try:
+        compiler_name = get_compiler_name(compiler_val)
+        if compiler_name == 'other':
+            return False
+        cmd = "echo __STDC_VERSION__ __STDC_NO_ATOMICS__ | "
+        cmd += compiler_name + " -E -x c - | sed -e '/^\#/d'"
+        out = run_command(['sh', '-c', cmd]).split()
```

Comment:
This seems wonky/fragile to me. We should be able to use a subprocess pipe instead of just creating up a shell command. We should also just be able to process the output with python instead of using sed